### PR TITLE
docs: removed broken link to Ambassador

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR removes a broken link to previous Ambassador content that was hosted.

### CODE REVIEW

File: website/data/docs-nav-data.json [hunk 1]
```diff
@@ -458,10 +458,6 @@
             "title": "Ingress Controllers",
             "path": "k8s/connect/ingress-controllers"
           },
-          {
```
Comment (david-yu) [inline website/data/docs-nav-data.json:461]: This one should stay since this link is not broken.

File: website/data/docs-nav-data.json [hunk 2]
```diff
@@ -458,10 +458,6 @@
...
-            "title": "Ambassador Edge Stack Integration",
-            "href": "https://learn.hashicorp.com/tutorials/consul/load-balancing-ambassador"
-          },
```
Comment (david-yu) [inline website/data/docs-nav-data.json:464]: This should remain since this link is not broken.

Comment (karl-cardenas-coding) [reply website/data/docs-nav-data.json:464]: I see, so you still want the link to the learn tutorial. Gotcha, I thought this was simply missed during the cleanup.

Comment (david-yu) [reply website/data/docs-nav-data.json:464]: Yes there was a previous outdated Ambassador doc that was severely outdated. I just forgot to remove it from nav-data as well. Thanks!

**Review by david-yu — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/498225.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 6880551c211727a606fc8a79b2bb18c6567524b4 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 6880551c211727a606fc8a79b2bb18c6567524b4 onto `release/1.10.x` succeeded!

**Pull request merged**